bluecellulab/stimulus/factory.py

```diff
@@ -687,3 +371,213 @@ def sinespec(
             )
 
         raise TypeError("You have to give either threshold_current or amplitude")
+
+    def ornstein_uhlenbeck(
+        self,
+        duration: float,
+        tau: float,
+        sigma: Optional[float] = None,
+        mean: Optional[float] = None,
+        mean_percent: Optional[float] = None,
+        sd_percent: Optional[float] = None,
+        threshold_current: Optional[float] = None,
+        seed: Optional[int] = None
+    ) -> Stimulus:
+        """Creates an Ornstein-Uhlenbeck process stimulus (factory-compatible).
+
+        Args:
+            duration: Duration of the stimulus (ms).
+            tau: Time constant of the noise process.
+            sigma: Standard deviation of the noise (used when mean is provided).
+            mean: Absolute mean current value (used if provided).
+            mean_percent: Mean current as a percentage of threshold current (used if mean is None).
+            sd_percent: Standard deviation as a percentage of threshold current (used if sigma is None).
+            threshold_current: Reference threshold current for percentage-based calculation.
+            seed: Optional random seed for reproducibility.
+
+        Returns:
+            A `Stimulus` object (OrnsteinUhlenbeck) that can be plotted and injected.
+
+        Notes:
+            - If `mean` is provided, `mean_percent` is ignored.
+            - If `threshold_current` is not provided, threshold-based parameters cannot be used.
+        """
+        is_amplitude_based = mean is not None and sigma is not None
+        is_threshold_based = (
+            threshold_current is not None
+            and threshold_current != 0
+            and mean_percent is not None
+            and sd_percent is not None
+        )
+
+        if is_amplitude_based:
+            if is_threshold_based:
+                logger.info(
+                    "mean, threshold_current, and mean_percent are all set in Ornstein-Uhlenbeck."
+                    " Using mean and ignoring threshold-based parameters."
+                )
+
+            return OrnsteinUhlenbeck.amplitude_based(
+                dt=self.dt,
+                duration=duration,
+                tau=tau,
+                sigma=sigma,  # type: ignore[arg-type]
+                mean=mean,  # type: ignore[arg-type]
+                seed=seed
+            )
+
+        if is_threshold_based:
+            return OrnsteinUhlenbeck.threshold_based(
+                dt=self.dt,
+                duration=duration,
+                mean_percent=mean_percent,  # type: ignore[arg-type]
+                sd_percent=sd_percent,  # type: ignore[arg-type]
+                threshold_current=threshold_current,  # type: ignore[arg-type]
+                tau=tau,
+                seed=seed
+            )
+
+        raise TypeError("You have to give either `mean` or `threshold_current`.")
```

Comment:
technically, have to give either mean and sigma or threshold_current and mean_percent and sd_percent